### DISCUSSION TITLE
Config question

### DISCUSSION THREAD

**Discussion by AgricolaJKB** [directus/directus#3904] [Archive]
Hey guys.

I am more of an programming amateure and perhaps my problem/question is pretty obvious - but perhaps someone can help. 

I installed directus on my server. I don't want the webapp on my main url, but on "/directus". I have set the rewriteRules to redirect requests with this pattern to the directus server hearing on port 8055. That seems to work, the server is starting normally and the request is passed to directus. 

But if I try to access directus with a browser, I get this error: {"errors":[{"message":"Route /directus doesn't exist.","extensions":{"code":"ROUTE_NOT_FOUND"}}]}

Apparently directus assumes, that the extension /directus is an API-endpoint. I changed the public_url in .env from "/" to "/directus" and "{myDomain}/directus", but both versions didn't help. I am still getting this error message when trying to access directus.

Do you know how to fix this problem?

I am using npx and Mariadb 10.3.

Thanks a lot!

P.s.: thanks to @benhaynes for the redirect! :)

**Comment by AgricolaJKB** [comment]
Fixed - needed just a little change in the rewriteRules. Ty though :)

**Reply by JakubOlejnik1155** [reply]
@AgricolaJKB I have the same problem. What was Your solution?

**Reply by kitschpatrol** [reply]
Hmm I'm also running into this same error trying to run Directus under a specific path on a multi-app Platform.sh deployment...

**Reply by kitschpatrol** [reply]
I think I got it working. Described solution in more detail here: https://github.com/platformsh-templates/directus/issues/40#issuecomment-974750037

**Comment by cloudpotions** [comment]
I realize this is an old post but wanted to help out others getting started with Directus as this one came up in top search results when I searched

{"errors":[{"message":"Route /defaultsite doesn't exist.","extensions":{"path":"/defaultsite","code":"ROUTE_NOT_FOUND"}}]}

I just pushed a commit to fix the Coolify template on Directus which had a similar problem (In a Traefik environment). So I changed it to have the /admin path and make sure that it was on port 8055. 

I found the solution, but read below to understand more. 

Here was the problem, I got this exact error when I tried to load the site in my browser I am working on (Chrome) but when I went incognito mode, Directus pulled up properly - so keep in mind, you might have to clear cache and cookies in your browser, and always test your routes incognito mode! 

For example, this is the only change I made in the Docker-Compose File that fixed it
The regular Web App Url (Coolify uses magic variables to make a .slip domain, but this will still address the fix in multiple scenarios)

$SERVICE_FQDN_DIRECTUS_8055

In coolify this wold automatically generate a random .slip url, something like http://directus-rwck8wsogc.123.456.789.sslip.io:8055

Problem, when you use that domain, it does not work. 

Fix - add the admin path to your URL environment variable and then it will rout the URL.com:8055 --> url.com/admin and then it will pull up the proper login screen 

example:  $SERVICE_FQDN_DIRECTUS_8055=/admin
